src/java/org/apache/cassandra/db/memtable/DefaultMemtableFactory.java

```diff
@@ -18,12 +18,59 @@
 
 package org.apache.cassandra.db.memtable;
 
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.cassandra.db.commitlog.CommitLogPosition;
+import org.apache.cassandra.db.partitions.PartitionUpdate;
+import org.apache.cassandra.db.partitions.TriePartitionUpdate;
+import org.apache.cassandra.metrics.TableMetrics;
+import org.apache.cassandra.schema.TableMetadataRef;
+
 /**
  * This class exists solely to avoid initialization of the default memtable class.
  * Some tests want to setup table parameters before initializing DatabaseDescriptor -- this allows them to do so.
  */
-public class DefaultMemtableFactory
+public class DefaultMemtableFactory implements Memtable.Factory
```

Comment:
It seems I forgot in CNDB-9850 that the OSS branch has a better solution to the problem this is meant to solve: extract the factory outside the memtable class, so that it no longer has a dependency on the memtable class being constructed. On the OSS branch that's the `SkipListMemtableFactory`, here we can do the same moving `TrieMemtable.Factory` to top level as `TrieMemtableFactory`, and then referencing its `INSTANCE` from the default params no longer involves initializing `TrieMemtable`.

This avoids having to reproduce every method of `Memtable.Factory` here which we can easily forget to do.

Reply:
Ok, I'll do that